packages/firestore/src/platform_browser/webchannel_connection.ts

```diff
@@ -371,10 +369,9 @@ export class WebChannelConnection implements Connection {
           // compatible with the bug we need to check either condition. The latter
           // can be removed once the fix has been rolled out.
           // Use any because msgData.error is not typed.
-          // eslint-disable-next-line @typescript-eslint/no-explicit-any
-          const msgDataAsAny: any = msgData;
+          const msgDataAsAny: WebChannelError | object = msgData;
```

Comment:
The name `msgDataAsAny` did not survive your refactor very well.  `msgDataOrError` ?

Reply:
Sounds good. Renamed.